src/hazelcore/TyVar.re

```diff
@@ -0,0 +1,30 @@
+open Sexplib.Std;
+
+[@deriving sexp]
+type t = string;
+
+let length: t => int = String.length;
+
+let equal: (t, t) => bool = String.equal;
+
+let valid_name: string => bool = {
+  let re = Re.Str.regexp("^[_a-zA-Z][_a-zA-Z0-9']*$");
+  s => Re.Str.string_match(re, s, 0);
+};
+
+let builtin_type: string => bool =
+  fun
+  | "Int"
+  | "Float"
+  | "Bool" => true
+  | _ => false;
+
+let keyword: string => bool =
+  fun
+  | "let"
```

Comment:
don't duplicate all the keywords here, use the ExpandingKeyword module

Reply:
Roger.

There are two equivalent functions in ExpandingKeyword.re: `of_string` is a simple fun on strings and `mk` is an equivalent series of if statements that calls the various `is_*` predicates.

Just dropping `mk` would be very easy.

Reply:
> don't duplicate all the keywords here, use the ExpandingKeyword module

fixed by e0da4528d